draft-dekater-scion-dataplane.md

```diff
@@ -1806,14 +1805,82 @@ However, the path choice of the endpoint may possibly be exploited by an attacke
+
+## SIG Framing
+
+IP packets are encapsulated over SCION/UDP into SIG frames. There may be multiple IP packets in a single SIG frame, and a single IP packet may be split into multiple SIG frames.
+
+A pair of SIGs establishes a SIG tunneling session. TODO: how?.  Within each session there may be multiple streams which is useful to distinguish between traffic sent by different SIG instances. For example, if an ingress SIG is restarted, it will create a new Stream ID for each session so the egress SIG will know that the new frame with a new Stream ID does not carry trailing part of the unfinished IP packet from a different stream.
+
+Each SIG frame has a sequence number is used by the egress SIG to reassemble the encapsulated IP packets within a stream.
+
+~~~~
+
++-----------------------+
+|         SCION         |
++-----------------------+
+|          UDP          |
++-----------------------+
+|    SIG frame header   |
++-----------------------+
+|   SIG frame payload   |
++-----------------------+
+
+~~~~
+{: #figure-28 title="SIG framing within a SCION packet"}
+
+## SIG Frame Header
+
+~~~~
+
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|     Version   |    Session    |            Index              |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|         Reserved        |               Stream                |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                       Sequence Number                         +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+~~~~
+{: #figure-29 title="SIG Frame Header format"}
+
+
+All fields within SIG Frame Header are in network byte order.
+
+- `Version` (8 bits) indicates the SIG framing version. It MUST be set to zero.
+- `Session` (8 bits) identifies a tunneling session between a pair of SIGs.
+- `Index` (16 bits) is the byte offset of the first beginning of an IP packet within the payload. If no IP packet starts in the payload, e.g. if the frame contains only a trailing part of an IP packet, the field MUST be set to 0xFFFF.
+- `Reserved` (12 bits): it MUST be set to zero.
+- `Stream` (20 bits), along with the session, it identifies a unique sequence of SIG frames. Frames from the same stream are, on the remote SIG, put into the same reassembly queue. There may be multiple streams per session.
+- `Sequence Number` (64 bits) indicates the position of the frame within a stream. Consecutive frames can be used to reassemble IP packets split among multiple frames.
+
+TODO:
+- how is the session established and teared down?
```

Comment:
Me no speak English good, but... "established and teared down?": isn't that supposed to be "established and torn down"?

Reply:
'established and torn down' would be correct.

Reply:
Thanks for catching i. It is a TODO anyways, where can we find something more  about this?